src/pages/changeProfile/index.ts

```diff
@@ -0,0 +1,204 @@
+      size: "small",
+      value: this.props.phone,
+      events: {
+        blur: (e: Event) => this.validate(e, (this.children.inputPhone as Block).setProps),
+        focus: (e: Event) => this.validate(e, (this.children.inputPhone as Block).setProps)
+      }
+    });
+  }
+
+  validate(event: Event, setProps: (nextProps: any) => void) {
+
+    const target = event.target as HTMLInputElement;
+    const field = target.name;
+
+    if (event.type === "blur") {
+      if (!this.isValid(target.value, field)) {
+
+        setProps({ error: true });
+
+        this.setDisableButton(true);
+      }
+    }
+
+    if (event.type === "focus") {
+      if (target.classList.contains("invalid")) {
+
+        setProps({ error: false });
+
+        this.setDisableButton(false);
+      }
+    }
+  }
+
+  setDisableButton(isDisable: boolean) {
+    (this.children.buttonSave as Block).setProps({ disabled: isDisable });
+  }
+
+  isValid(value: string, field: string) {
+    switch (field) {
+      case "login":
+        return validationPatterns.login.test(value);
+      case "email":
+        return validationPatterns.email.test(value);
+      case "first_name":
+      case "second_name":
+        return validationPatterns.username.test(value);
+      case "phone":
+        return validationPatterns.phone.test(value);
+    }
+  }
+
+  onSubmit() {
+    const form = this.getForm();
+
+    const isEmailValid = this.isValid(form.email as string, "email");
+    const isLoginValid = this.isValid(form.login as string, "login");
+    const isFirstNameValid = this.isValid(form.firstName as string, "first_name");
+    const isSecondNameValid = this.isValid(form.secondName as string, "second_name");
+    const isPhoneValid = this.isValid(form.phone as string, "phone");
+
+    if (isLoginValid && isEmailValid && isFirstNameValid && isSecondNameValid && isPhoneValid) {
+      console.log("formData", form);
+    }
+
+    if (!isEmailValid) {
+      (this.children.inputPost as Block).setProps({ error: true });
+    }
+    if (isEmailValid) {
+      (this.children.inputPost as Block).setProps({ error: false });
+    }
+
+    if (!isLoginValid) {
+      (this.children.inputLogin as Block).setProps({ error: true });
+    }
+    if (isLoginValid) {
+      (this.children.inputLogin as Block).setProps({ error: false });
+    }
+
+    if (!isFirstNameValid) {
+      (this.children.inputFirstName as Block).setProps({ error: true });
+    }
+    if (isFirstNameValid) {
+      (this.children.inputFirstName as Block).setProps({ error: false });
+    }
+
+    if (!isSecondNameValid) {
+      (this.children.inputSecondName as Block).setProps({ error: true });
+    }
+    if (isFirstNameValid) {
+      (this.children.inputSecondName as Block).setProps({ error: false });
+    }
+    if (!isPhoneValid) {
+      (this.children.inputPhone as Block).setProps({ error: true });
+    }
+    if (isPhoneValid) {
+      (this.children.inputPhone as Block).setProps({ error: false });
+    }
```

Comment:
Думаю, что эту запись можно сократить
```suggestion
   this.children.inputPhone as Block).setProps({ error: !isPhoneValid });
```
C другими полями по аналогии

Reply:
done